src/core/gsthread.cpp

```diff
@@ -1056,7 +1099,44 @@ void GraphicsSynthesizerThread::draw_pixel(int32_t x, int32_t y, uint32_t z, RGB
     if (!pass_depth_test)
         return;
 
-    uint32_t frame_color = read_PSMCT32_block(current_ctx->frame.base_pointer, current_ctx->frame.width, x, y);
+    uint32_t frame_color = 0;
+    switch (current_ctx->frame.format)
+    {
+        case 0x0:
+            frame_color = read_PSMCT32_block(current_ctx->frame.base_pointer, current_ctx->frame.width, x, y);
+            break;
+        case 0x1://24
+            frame_color = read_PSMCT32_block(current_ctx->frame.base_pointer, current_ctx->frame.width, x, y);
+            break;
+        case 0x2:
+            frame_color = convert_color_up(read_PSMCT16_block(current_ctx->frame.base_pointer, current_ctx->frame.width, x, y));
+            break;
+        case 0xA:
+            frame_color = convert_color_up(read_PSMCT16S_block(current_ctx->frame.base_pointer, current_ctx->frame.width, x, y));
+            break;
+        case 0x13:
+        case 0x14:
+        case 0x1B:
+        case 0x1C:
+        case 0x2C:
+            Errors::die("Reserved FRAME format (%d) read attempted", current_ctx->frame.format);
```

Comment:
These extra case statements are unnecessary given that the program already terminates with an unknown format.